modules/ruxfdtable/src/lib.rs

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) [2023] [Syswonder Community]
+ *   [Ruxos] is licensed under Mulan PSL v2.
+ *   You can use this software according to the terms and conditions of the Mulan PSL v2.
+ *   You may obtain a copy of Mulan PSL v2 at:
+ *               http://license.coscl.org.cn/MulanPSL2
+ *   THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+ *   See the Mulan PSL v2 for more details.
+ */
+
+//! fd table and FileLike trait for file system
+#![no_std]
+extern crate alloc;
+use alloc::sync::Arc;
+use core::marker::Send;
+use core::marker::Sync;
+
+use axerrno::LinuxResult;
+use axio::PollState;
+use flatten_objects::FlattenObjects;
+use spin::RwLock;
+
+///Rust version for struct timespec in ctypes. Represents a high-resolution time specification.
+pub struct RuxTimeSpec {
+    /// Whole seconds part of the timespec.
+    pub tv_sec: core::ffi::c_longlong,
+    /// Nanoseconds part of the timespec, complementing `tv_sec`.
+    pub tv_nsec: core::ffi::c_long,
+}
+
+///Rust version for struct stat in ctypes. Represents file status information.
+#[cfg(target_arch = "aarch64")]
+pub struct RuxStat {
```

Comment:
A question about the struct `RuxStat`, why this variable need different version for different arch? Another underlying problem is that the naming of `RuxStat` might be a little board.

Reply:
I think it is possible to use `#[cfg(...)]` inside the struct rather than two standalone defination if the defination if a little difference exists.

Reply:
Some of the fields, like `st_nlink`'s type is different, and some fields' order is different. So here two different structs are maintained. But it is possible to make it concise: use `#[cfg(...)]` for `st_nlink` and `__unused`, re-order the fields when transferring to C struct.

Reply:
> Some of the fields, like `st_nlink`'s type is different, and some fields' order is different. So here two different structs are maintained. But it is possible to make it concise: use `#[cfg(...)]` for `st_nlink` and `__unused`, re-order the fields when transferring to C struct.

Thanks, I understand now. If the difference is the order of field, using `#[cfg(...)]` might make code messy. I think it will be okay to keep the code as it is now.